tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/contact/openpaas/OpenPaasContactsConsumer.java

```diff
@@ -0,0 +1,80 @@
+package com.linagora.tmail.james.jmap.contact.openpaas;
+
+import jakarta.inject.Inject;
+
+import org.apache.james.backends.rabbitmq.ReactorRabbitMQChannelPool;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linagora.tmail.james.jmap.contact.TmailContactUserAddedEvent;
+
+import java.io.IOException;
+import reactor.core.Disposable;
+import scala.NotImplementedError;
+
+
+public class OpenPaasContactsConsumer {
+    private final Logger LOGGER = LoggerFactory.getLogger(OpenPaasContactsConsumer.class);
+
+    private static final String QUEUE_NAME = "contacts:contact:add";
+    private final ReactorRabbitMQChannelPool channelPool;
+    private Disposable consumeContactsDisposable;
+
+    // TODO: Create a separate RabbitMQ module for OpenPaaS communication so the injected channel pool
+    //  would be custom configured
```

Comment:
I support this! A Lot!

We could...

Have a file named `openpaas.properties` which allow overriding RABBITMQ connection settings, if absent we can fallback to the default ChannelPool. It would be annotated with @Named("openpaas")

Then we can use it for OpenPaaS related communications in particular have a `OpenPaaSAMQPForwardAttribute` that extends `AmqpForwardAttribute and reuses this.

Actually such a task might be a good prelimiary worrk to this PR!

Reply:
https://github.com/linagora/tmail-backend/issues/1223